src/hotspot/share/compiler/methodMatcher.cpp

```diff
@@ -239,6 +238,8 @@ void skip_leading_spaces(char*& line, int* total_bytes_read ) {
   }
 }
 
+enum OptionType parse_option_type(const char* type_str);
```

Comment:
Hmm, rather than a surprising forward declaration to a method declared in compilerOracle.cpp (does this actually work?) maybe declare that method publically as `CompilerOracle::parse_option_type` and call that?

Reply:
yes, it's a forward declaration.  I guess some recursive headers include compilerOracle.hpp(precompiled.hpp?) I will declare it as a public static member function like `CompilerOracle::string_to_option`